IOS: type 'BUAdSDKManager' has no member 'setIsPaidApp'

**1.简述问题**

在一个新项目中使用 `pangle_flutter` 无法启动，完整错误信息如下：

```
Launching lib/main.dart on iPhone 13 in debug mode...
Xcode build done.                                           14.6s
Failed to build iOS app
Error output from Xcode build:
↳
    --- xcodebuild: WARNING: Using the first of multiple matching destinations:
    { platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
    { platform:iOS Simulator, id:6DF789CD-1E03-4B77-8F74-E1E992A2F9A6, OS:15.0, name:iPhone 13 }
    { platform:iOS Simulator, id:5DBF6BCF-928C-46AB-99A2-25DCC2ECBAFE, OS:15.0, name:iPhone 13 Pro }
    { platform:iOS Simulator, id:8DEFF211-09F5-4337-B187-0465B09C4ED4, OS:15.0, name:iPhone 13 mini }
    { platform:iOS Simulator, id:5CCE74B8-4ABD-4E3E-9E22-0AC13555BB38, OS:15.0, name:iPhone SE (2nd generation) }
    { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
    ** BUILD FAILED **
Xcode's output:
↳
    /Users/cipchk/.pub-cache/hosted/pub.flutter-io.cn/pangle_flutter-1.4.7/ios/Classes/PangleAdManager.swift:48:28: error: type 'BUAdSDKManager' has no member 'setIsPaidApp'
                BUAdSDKManager.setIsPaidApp(isPaidApp!)
                ~~~~~~~~~~~~~~ ^~~~~~~~~~~~
    /Users/cipchk/.pub-cache/hosted/pub.flutter-io.cn/pangle_flutter-1.4.7/ios/Classes/PangleAdManager.swift:56:42: error: cannot convert value of type 'UInt' to expected argument type 'Int'
                BUAdSDKManager.setCoppa(coppa!)
                                             ^
                                        Int(  )
    /Users/cipchk/Library/Developer/Xcode/DerivedData/Runner-ewjykgsqdascxtdttfmoiryshgtl/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/pangle_flutter.build/Objects-normal/x86_64/FLTTaskProtocol.dia:1:1: warning: Could not read serialized diagnostics file: error("Invalid diagnostics signature") (in target 'pangle_flutter' from project 'Pods')
    Command CompileSwift failed with a nonzero exit code
    /Users/cipchk/.pub-cache/hosted/pub.flutter-io.cn/pangle_flutter-1.4.7/ios/Classes/PangleAdManager.swift:48:28: error: type 'BUAdSDKManager' has no member 'setIsPaidApp'
                BUAdSDKManager.setIsPaidApp(isPaidApp!)
                ~~~~~~~~~~~~~~ ^~~~~~~~~~~~
    /Users/cipchk/.pub-cache/hosted/pub.flutter-io.cn/pangle_flutter-1.4.7/ios/Classes/PangleAdManager.swift:56:42: error: cannot convert value of type 'UInt' to expected argument type 'Int'
                BUAdSDKManager.setCoppa(coppa!)
                                             ^
                                        Int(  )
    /Users/cipchk/.pub-cache/hosted/pub.flutter-io.cn/pangle_flutter-1.4.7/ios/Classes/PangleAdManager.swift:48:28: error: type 'BUAdSDKManager' has no member 'setIsPaidApp'
                BUAdSDKManager.setIsPaidApp(isPaidApp!)
                ~~~~~~~~~~~~~~ ^~~~~~~~~~~~
    /Users/cipchk/.pub-cache/hosted/pub.flutter-io.cn/pangle_flutter-1.4.7/ios/Classes/PangleAdManager.swift:56:42: error: cannot convert value of type 'UInt' to expected argument type 'Int'
                BUAdSDKManager.setCoppa(coppa!)
                                             ^
                                        Int(  )
    /Users/cipchk/Library/Developer/Xcode/DerivedData/Runner-ewjykgsqdascxtdttfmoiryshgtl/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/pangle_flutter.build/Objects-normal/arm64/FLTTaskProtocol.dia:1:1: warning: Could not read serialized diagnostics file: error("Invalid diagnostics signature") (in target 'pangle_flutter' from project 'Pods')
    Command CompileSwift failed with a nonzero exit code
    /Users/cipchk/.pub-cache/hosted/pub.flutter-io.cn/pangle_flutter-1.4.7/ios/Classes/PangleAdManager.swift:48:28: error: type 'BUAdSDKManager' has no member 'setIsPaidApp'
                BUAdSDKManager.setIsPaidApp(isPaidApp!)
                ~~~~~~~~~~~~~~ ^~~~~~~~~~~~
    /Users/cipchk/.pub-cache/hosted/pub.flutter-io.cn/pangle_flutter-1.4.7/ios/Classes/PangleAdManager.swift:56:42: error: cannot convert value of type 'UInt' to expected argument type 'Int'
                BUAdSDKManager.setCoppa(coppa!)
                                             ^
                                        Int(  )
    /Users/cipchk/Desktop/flutter-demo/ios/Pods/libwebp/src/enc/backward_references_enc.c:694:12: warning: variable 'best_offset' may be uninitialized when used here [-Wconditional-uninitialized]
              (best_offset << MAX_LENGTH_BITS) | (uint32_t)best_length;
               ^~~~~~~~~~~
    /Users/cipchk/Desktop/flutter-demo/ios/Pods/libwebp/src/enc/backward_references_enc.c:626:20: note: initialize the variable 'best_offset' to silence this warning
        int best_offset;
                       ^
                        = 0
    1 warning generated.
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in parallel
    /Users/cipchk/Desktop/flutter-demo/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'libwebp' from project 'Pods')
    /Users/cipchk/Desktop/flutter-demo/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'Mantle' from project 'Pods')
    /Users/cipchk/Desktop/flutter-demo/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'FMDB' from project 'Pods')
    /Users/cipchk/Desktop/flutter-demo/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'AlipaySDK-iOS' from project 'Pods')
Could not build the application for the simulator.
Error launching application on iPhone 13.
Exited (sigterm)
```

**2.重现步骤**

- `flutter create flutter-demo`
- 在 `pubspec.yaml` 增加 `pangle_flutter: ^1.4.7`
- 在 `main.dart` 增加以下代码
```dart
@override
void initState() {
  super.initState();
  testInit();
}

Future testInit() async {
  await pangle.init(
    iOS: const IOSConfig(appId: "kAppId"),
    android: const AndroidConfig(appId: "kAppId"),
  );
}
```

**3.期望结果**


**4.SDK版本:**
- Flutter 2.5.3 • channel stable • https://github.com/flutter/flutter.git
- Framework • revision 18116933e7 (6 weeks ago) • 2021-10-15 10:46:35 -0700
- Engine • revision d3ea636dc5
- Tools • Dart 2.14.4
 - pangle_flutter: 1.4.7
 - iOS: 15.0

**额外信息**
无

--- a/README.md
+++ b/README.md
@@ -131,6 +131,7 @@ dependencies:
         symlink = File.join(symlink_plugins_dir, plugin_name)
         File.symlink(plugin_path, symlink)
 
+        pod plugin_name, :path => File.join('.symlinks', 'plugins', plugin_name, 'ios')
         if plugin_name == 'pangle_flutter'
           # cn表示国内，global表示海外
           pod 'pangle_flutter/global', :path => File.join('.symlinks', 'plugins', plugin_name, 'ios')

--- a/example/lib/page/home_page_provider.dart
+++ b/example/lib/page/home_page_provider.dart
@@ -46,51 +46,53 @@ mixin HomePageProviderStateMixin<T extends StatefulWidget> on State<T> {
       ),
       body: Padding(
         padding: EdgeInsets.all(16),
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.start,
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: <Widget>[
-            ListTile(
-              title: Text('Testing environment:'),
-              subtitle: Padding(
-                padding: const EdgeInsets.only(top: 8.0),
-                child: Text(kEnv),
+        child: SingleChildScrollView(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.start,
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              ListTile(
+                title: Text('Testing environment:'),
+                subtitle: Padding(
+                  padding: const EdgeInsets.only(top: 8.0),
+                  child: Text(kEnv),
+                ),
               ),
-            ),
-            ListTile(
-              title: Text('Dependencies:'),
-              subtitle: Padding(
-                padding: const EdgeInsets.only(top: 8.0),
-                child: Text(_denpendencies ?? ''),
+              ListTile(
+                title: Text('Dependencies:'),
+                subtitle: Padding(
+                  padding: const EdgeInsets.only(top: 8.0),
+                  child: Text(_denpendencies ?? ''),
+                ),
               ),
-            ),
-            ListTile(
-              title: Text('Theme: '),
-              subtitle: Padding(
-                padding: const EdgeInsets.only(top: 8.0),
-                child: Text('$_theme'),
+              ListTile(
+                title: Text('Theme: '),
+                subtitle: Padding(
+                  padding: const EdgeInsets.only(top: 8.0),
+                  child: Text('$_theme'),
+                ),
               ),
-            ),
-            ElevatedButton(
-              onPressed: requestPermissions,
-              child: Text('Request Permissions'),
-            ),
-            SizedBox(height: 30),
-            ElevatedButton(
-              onPressed: changeTheme,
-              child: Text('Change Theme'),
-            ),
-            SizedBox(height: 30),
-            ElevatedButton(
-              onPressed: loadNativeAd,
-              child: Text('Native AD'),
-            ),
-            SizedBox(height: 30),
-            ElevatedButton(
-              onPressed: loadExpressAd,
-              child: Text('Express AD'),
-            ),
-          ],
+              ElevatedButton(
+                onPressed: requestPermissions,
+                child: Text('Request Permissions'),
+              ),
+              SizedBox(height: 30),
+              ElevatedButton(
+                onPressed: changeTheme,
+                child: Text('Change Theme'),
+              ),
+              SizedBox(height: 30),
+              ElevatedButton(
+                onPressed: loadNativeAd,
+                child: Text('Native AD'),
+              ),
+              SizedBox(height: 30),
+              ElevatedButton(
+                onPressed: loadExpressAd,
+                child: Text('Express AD'),
+              ),
+            ],
+          ),
         ),
       ),
     );

--- a/ios/Classes/FLTSplashView.swift
+++ b/ios/Classes/FLTSplashView.swift
@@ -80,8 +80,8 @@ class SplashView: UIView {
         self.methodChannel = methodChannel
         let expressArgs = params["expressSize"] as? [String: Double] ?? [:]
         let uiframe = UIScreen.main.bounds
-        let width = expressArgs["width"] ?? uiframe.width
-        let height = expressArgs["height"] ?? uiframe.height
+        let width: Double = expressArgs["width"] ?? Double(uiframe.width)
+        let height: Double = expressArgs["height"] ?? Double(uiframe.height)
         let frame = CGRect(x: 0, y: 0, width: width, height: height)
         super.init(frame: frame)
 

--- a/ios/Classes/PangleAdManager.swift
+++ b/ios/Classes/PangleAdManager.swift
@@ -40,21 +40,14 @@ public final class PangleAdManager: NSObject {
         let appId: String = args["appId"] as! String
         let logLevel: Int? = args["logLevel"] as? Int
         let coppa: UInt? = args["coppa"] as? UInt
-        let isPaidApp: Bool? = args["isPaidApp"] as? Bool
 
         BUAdSDKManager.setAppID(appId)
-
-        if isPaidApp != nil {
-            BUAdSDKManager.setIsPaidApp(isPaidApp!)
-        }
 
         if logLevel != nil {
             BUAdSDKManager.setLoglevel(BUAdSDKLogLevel(rawValue: logLevel!)!)
         }
 
-        if coppa != nil {
-            BUAdSDKManager.setCoppa(coppa!)
-        }
+        BUAdSDKManager.setCoppa(((coppa!=nil) && (coppa! > 0)) ? 1 : 0)
     }
 
     public func loadSplashAd(_ args: [String: Any?], result: @escaping FlutterResult) {

--- a/ios/Classes/PangleAdManager.swift
+++ b/ios/Classes/PangleAdManager.swift
@@ -47,7 +47,7 @@ public final class PangleAdManager: NSObject {
             BUAdSDKManager.setLoglevel(BUAdSDKLogLevel(rawValue: logLevel!)!)
         }
 
-        BUAdSDKManager.setCoppa(((coppa!=nil) && (coppa! > 0)) ? 1 : 0)
+        BUAdSDKManager.setCoppa(((coppa != nil) && (coppa! > 0)) ? 1 : 0)
     }
 
     public func loadSplashAd(_ args: [String: Any?], result: @escaping FlutterResult) {

